os/lib/ccm-star.c

```diff
@@ -152,9 +151,7 @@ aead(const uint8_t* nonce,
     uint8_t *result, uint8_t mic_len,
     int forward)
 {
-  if(!MIC_LEN_VALID(mic_len)) {
-    return;
-  }
+  assert(mic_len <= AES_128_BLOCK_SIZE);
```

Comment:
Since code might not be compiled with assertions enabled, this changes the semantics. Are there external callers to this function?

Reply:
As the previous version only worked when passing standards-compliant MIC lengths, nothing should break when we expand the valid range.

Reply:
What happens with a faulty length in code built without assertions, does Contiki-NG crash?

Reply:
I am going to create another PR where I change the return type to int so that we can bubble up errors.